### PR TITLE
refactor(server): download file

### PR DESCRIPTION
**Pull request by jrasm91**

### CODE REVIEW

File: server/apps/immich/src/api-v1/asset/asset.controller.ts [hunk 1]
```diff
@@ -103,12 +108,9 @@ export class AssetController {
   async downloadFile(
     @GetAuthUser() authUser: AuthUserDto,
     @Response({ passthrough: true }) res: Res,
-    @Query(new ValidationPipe({ transform: true })) query: ServeFileDto,
     @Param('assetId') assetId: string,
   ): Promise<any> {
-    this.assetService.checkDownloadAccess(authUser);
-    await this.assetService.checkAssetsAccess(authUser, [assetId]);
-    return this.assetService.downloadFile(query, assetId, res);
+    return this.assetService.downloadFile(authUser, assetId).then(asStreamableFile);
```
Comment (alextran1502) [inline server/apps/immich/src/api-v1/asset/asset.controller.ts:113]: I find this a bit hard to follow. Can you elaborate for documentation?

Comment (alextran1502) [reply server/apps/immich/src/api-v1/asset/asset.controller.ts:113]: Ah, downloadFile return `ImmichStream` and you cast that to StreamableFile of Nestjs

File: server/apps/immich/src/api-v1/asset/asset.service.ts [hunk 2]
```diff
@@ -189,62 +196,21 @@ export class AssetService {
...
-  public async downloadFile(query: ServeFileDto, assetId: string, res: Res) {
-    try {
-      let fileReadStream = null;
-      const asset = await this._assetRepository.getById(assetId);
-
-      // Download Video
-      if (asset.type === AssetType.VIDEO) {
-        const { size } = await fileInfo(asset.originalPath);
-
-        res.set({
```
Comment (alextran1502) [inline server/apps/immich/src/api-v1/asset/asset.service.ts:201]: I don't recal why I had to set the content-type and content-length here. But the testing went well even without these two header get explicitly set. Did you encounter anything not working right without the two headers get set?

**Review by alextran1502 — COMMENTED** [review]
Looks good! I tested on mobile, web and public shared page. All worked as expected

File: server/apps/immich/src/api-v1/asset/asset.service.ts (hunk 2 above)
Comment (jrasm91) [reply server/apps/immich/src/api-v1/asset/asset.service.ts:201]: So these actually do get set, just through the options in the streamable file object. I'm trying to keep transport specific stuff like this outside of the "domain". The domain returns the value, then the controller maps the response into a streamable file. The second argument ends up being set as a header.

File: server/apps/immich/src/api-v1/asset/asset.controller.ts (hunk 1 above)
Comment (jrasm91) [reply server/apps/immich/src/api-v1/asset/asset.controller.ts:113]: Could return a streamable file directly, but I'd prefer to keep that nestjs reference out of the domain, since it is specific to the api (not cli, graphql, etc.), so I made a separate interface with no nestjs specifics in it (immich readable file), and then map between the two here.

File: server/apps/immich/src/api-v1/asset/asset.service.ts (hunk 2 above)
Comment (alextran1502) [reply server/apps/immich/src/api-v1/asset/asset.service.ts:201]: image
Through this?

Comment (jrasm91) [reply server/apps/immich/src/api-v1/asset/asset.service.ts:201]: Yes
